src/util.js

```diff
@@ -185,3 +185,32 @@ export function findClosestRecursively(element, selector) {
     )
   }
 }
+
+export function elementIsFocusable(element) {
+  const inertDisabledOrHidden = "[inert], :disabled, [hidden], details:not([open]), dialog:not([open])"
+
+  return !!element && element.closest(inertDisabledOrHidden) == null && typeof element.focus == "function"
+}
+
+export function queryAutofocusableElement(elementOrDocumentFragment) {
+  for (const element of elementOrDocumentFragment.querySelectorAll("[autofocus]")) {
+    if (elementIsFocusable(element)) return element
+    else continue
+  }
+
+  return null
+}
+
+export async function around(callback, reader) {
+  const before = reader()
+  await waitForCallback(callback)
+  const after = reader()
+
+  return [before, after]
+}
+
+export function waitForCallback(callback) {
+  callback()
+
+  return nextAnimationFrame()
+}
```

Comment:
This is not awaiting for `callback` to finish. I think it works anyway because at the moment we're only using it with a non-async callbacks. But the net effect of this the same as:

```js
callback()
await nextAnimationFrame()

// carry on after nextAnimationFrame...
```

Is that what we are trying to achieve? If that's the case, I think we can inline those two lines. That removes some code indirection and makes more explicit what's going on.